src/main/java/org/truffleruby/cext/CExtNodes.java

```diff
@@ -873,20 +872,16 @@ public Object read(DynamicObject string, int index,
 
         @Specialization(guards = "isRubyString(string)")
         public int write(DynamicObject string, int index, int value,
-                @Cached("createBinaryProfile()") ConditionProfile nativeRopeProfile,
-                @Cached("getHelperNode()") StringNodes.SetByteNode setByteNode) {
+                @Cached("createBinaryProfile()") ConditionProfile newRopeProfile,
```

Comment:
It's a bit hard to read in GitHub, but this node looks nearly identical to `StringNodes.SetByteNode` from what I can tell. Can we make use of that?

Reply:
Not easily, we don't want the bound checks here.
`RSTRING_PTR(str)[i]` must be able to write beyond `bytesize`.

Reply:
Okay. Hard to see here. The call is simple enough, so I don't think we're losing anything by having similar logic in two places. It's the most natural way to call `SetByteNode`.